### DISCUSSION TITLE
npm global install on ubuntu based distribtions does not work out of the box

### DISCUSSION THREAD

**Discussion by PhilStantonCoding** [npm/feedback#925] [Registry]
I was dealing with a very annoying issue yesterday where I could not install npm packages globally. npm requires one to execute the command with root privileges to to install a package globally, in my case, it was angular packages. Every time I executed an "npm install -g"  as root, npm got stuck.

for example, I needed to install some devkits: 

npm install -g @angular-devkit/build-angular

returns an error, as it is trying to write to a directory that needs root privileges. 

Therefore, I run: 
sudo npm install -g @angular-devkit/build-angular 

but it gets immediately stuck on 
"IdealTree:lib" still idealTree buildDeps

and it was because there was not a .npmrc in the root users home directory, and it could not identify the registry URL to connect to. Could you consider adding this file the root users home directory in he future?

**Comment by ljharb** [comment]
There shouldn't need to be an npmrc in the user's home directory for the default registry - if you have a private one then you'd certainly need that, but it wouldn't be appropriate for npm to try to put it there.

**Reply by PhilStantonCoding** [reply]
I understand, as It could impose security risks, upon other things. Although, I was just able to reproduce the error by removing the config file from the root users home directory and immediately got stuck installing again. Maybe it has to do with the way I installed node and npm, but thanks for the reply
